datafusion/src/physical_plan/planner.rs

```diff
@@ -56,6 +56,121 @@ use expressions::col;
 use log::debug;
 use std::sync::Arc;
 
+fn create_function_physical_name(
+    fun: &str,
+    distinct: bool,
+    args: &[Expr],
+    input_schema: &DFSchema,
+) -> Result<String> {
+    let names: Vec<String> = args
+        .iter()
+        .map(|e| physical_name(e, input_schema))
+        .collect::<Result<_>>()?;
+
+    let distinct_str = match distinct {
+        true => "DISTINCT ",
+        false => "",
+    };
+    Ok(format!("{}({}{})", fun, distinct_str, names.join(",")))
+}
+
+fn physical_name(e: &Expr, input_schema: &DFSchema) -> Result<String> {
```

Comment:
This function is responsible for generating physical field name for a given expression based on https://github.com/apache/arrow-datafusion/blob/master/docs/specification/output-field-name-semantic.md.

I kept `Expr.name` as is for better readability in query explain and debug output.

Reply:
eventually maybe this might be easier to find if it were on `Expr` itself (e.g. `Expr::physical_name`)